main/realmain.cc

```diff
@@ -662,7 +674,7 @@ int realmain(int argc, char *argv[]) {
 #endif
     if (!gs || gs->hadCriticalError()) {
         returnCode = 10;
-    } else if (returnCode == 0 && gs->totalErrors() > 0 && !opts.supressNonCriticalErrors) {
+    } else if (returnCode == 0 && gs->errorQueue->nonSilencedErrorCount > 0 && !opts.supressNonCriticalErrors) {
```

Comment:
(Nit) Is this change needed? (`gs->totalErrors()` still exists right?)

Reply:
oh good call. Not sure why I changed this now